pedalboard/TimeStretch.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * pedalboard
+ * Copyright 2022 Spotify AB
+ *
+ * Licensed under the GNU Public License, Version 3.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "../vendors/rubberband/rubberband/RubberBandStretcher.h"
+
+using namespace RubberBand;
+
+namespace Pedalboard {
+
+/*
+ * A wrapper around Rubber Band that allows calling it independently of a plugin
+ * context, to allow for both pitch shifting and time stretching on fixed-size
+ * chunks of audio.
+ *
+ * The `Plugin` base class requires that one sample of audio output is always
+ * provided for every sample input, but this assumption does not hold true
+ */
+static juce::AudioBuffer<float>
+timeStretch(const juce::AudioBuffer<float> input, double sampleRate,
+            double speedFactor, double pitchShiftInSemitones) {
```

Comment:
nit: call this `stretchFactor`? Speed changes are clasically resampling, i.e. pitch and time at the same time, which is distinct from "stretch" which is only time.